rootfs/etc/nginx/lua/test/configuration_test.lua

```diff
@@ -184,6 +185,56 @@ describe("Configuration", function()
       assert.same(ngx.status, ngx.HTTP_BAD_REQUEST)
     end)
 
+    it("should not delete ocsp_response_cache if certificate remain the same", function()
+      ngx.shared.certificate_data.get = function(self, uid)
+        return "pemCertKey"
+      end
+
+      mock_ssl_configuration({
+        servers = { ["hostname"] = UUID },
+        certificates = { [UUID] = "pemCertKey" }
+      })
+
+      local s = spy.on(ngx.shared.ocsp_response_cache, "delete")
+      assert.has_no.errors(configuration.handle_servers)
+      assert.spy(s).was_not_called_with(UUID)
+    end)
+
+    it("should not delete ocsp_response_cache if certificate is empty", function()
+      ngx.shared.certificate_data.get = function(self, uid)
+          return nil
+      end
+
+      mock_ssl_configuration({
+        servers = { ["hostname"] = UUID },
+        certificates = { [UUID] = "pemCertKey" }
+      })
+
+      local s = spy.on(ngx.shared.ocsp_response_cache, "delete")
+      assert.has_no.errors(configuration.handle_servers)
+      assert.spy(s).was_not_called_with(UUID)
+    end)
```

Comment:
I don't understand how this test is passing. Looking at your implementation I'd expect this test to fail. Can you explain?

Reply:
Oops, i wrote three test cases at first but always failed at the third one. Finally I found out that the `ocsp_response_cache` take `(self, key)` as arguments, so I should use `assert.spy(s).was_not_called_with(ocsp_response_cache, UUID)` to satisfy busted.

But I forget to make up the arguments in the first two cases.

---

I fixed it use `assert.spy(s).was_not_called()` and added the `old_cert ~= nil` criteria in configuration.lua